internal/logs/workload_writer.go

```diff
@@ -0,0 +1,60 @@
+package logs
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/project-flotta/flotta-device-worker/internal/workload"
+)
+
+type WorkloadWriter struct {
+	workloadName string
+	cancel       context.CancelFunc
+	target       *FifoLog
+	targetName   string
+	manager      *workload.WorkloadManager
+}
+
+func NewWorkloadWriter(workload string, manager *workload.WorkloadManager) *WorkloadWriter {
+	res := &WorkloadWriter{
+		workloadName: workload,
+		manager:      manager,
+	}
+	return res
+}
+
+func (ww *WorkloadWriter) SetTarget(name string, target *FifoLog) {
+	ww.target = target
+	ww.targetName = name
+}
+
+func (ww *WorkloadWriter) Write(data []byte) (n int, err error) {
+	entry := NewLogEntry(data, ww.workloadName)
+	if ww.target == nil {
+		return 0, fmt.Errorf("cannot write to target")
```

Comment:
In this case the problem is that there is not FIFOLog available, WDYT to make the log a little clearer?